packages/plugin-ext-metrics/src/common/plugin-metrics-interfaces.ts

```diff
@@ -0,0 +1,80 @@
+/********************************************************************************
+ * Copyright (C) 2019 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+// Define common interfaces that multiple classes can use
+
+export interface MetricsMap {
```

Comment:
hello, shouldn't we use something like 
```
type MetricsMap = Map<string, MethodToAnalytics>;
```
I expect from a Map interface to have map functions

Reply:
The only reason I did it this way was that calling JSON.stringify on a map just returns '{}'. Instead of writing a function that converts the map into an object then converting that object into a string with JSON.stringify on the frontend and on the backend converting it back to an object then to a map, I just went with a javascript object so it can just be directly converted in and out of a string